### ISSUE TITLE
Incremental Font Transfer

### ISSUE THREAD

**Issue by omasanori** [mozilla/standards-positions#872]
## Request for Mozilla Position on an Emerging Web Specification

* Specification Title: Incremental Font Transfer
* Specification or proposal URL (if available): https://www.w3.org/TR/IFT/
* Explainer URL (if available): 
* Caniuse.com URL (optional): 
* Bugzilla URL (optional): 
* Mozillians who can provide input (optional): 

### Other information

[Progressive Font Enrichment: Evaluation Report](https://www.w3.org/TR/PFE-evaluation/ "Progressive Font Enrichment: Evaluation Report") provides some background information.

**Comment by svgeesus** [comment]
Adding additional information to this request for a standards position:

- Specification or proposal URL
  - Editors Draft: https://w3c.github.io/IFT/Overview.html
  - Published Specification (9 July 2024): https://www.w3.org/TR/IFT/ 
- Explainer URL: https://github.com/w3c/IFT/blob/main/IFT-Explainer.md
- Mozillians who can provide input: @jfkthame

**Comment by zcorpan** [comment]
@lars-t-hansen @valenting do you have opinions here? Should this be more general and part of the network stack?

**Comment by svgeesus** [comment]
> Should this be more general and part of the network stack?

Not understanding how a font-specific procedure (to create the initial subset font and the patches) could be made more general? The actual **transfer** part is just HTTP, nothing different there

**Comment by martinthomson** [comment]
Yes, this is about how we might source the content of `@font-face`, going from a single fetch (or maybe one for each codepoint range) to having the font rendering code the ability to initiate fetches as it discovers that it needs more information.  As long as this integrates with fetch properly (this cites https://www.w3.org/TR/css-fonts-4/#font-fetching-requirements which seems fine to me) that part should not affect networking at all.

That's one of the best features of this change: all the pieces are ordinary fetches, with ordinary caching rules.  They inherit some of the quirks of font fetching, but those are pretty well understood now.  I recall there being some connection reuse quirks involved, but I'm assuming that this is OK here: even if you need a new connection for a font, the incremental fetches can reuse that connection (assuming that the patch is on the same origin, I guess).

**Comment by svgeesus** [comment]
In general one would expect the initial font plus all the patches to be on the same origin. Remember that the patch urls are stored inside the font, so if the patches are moved to another location they won't work. Relative urls are fine, so we would expect to see these files exported from some tool and used together as a set of resources.
